packages/expo-router/src/modal/Modal.tsx

```diff
@@ -100,16 +100,17 @@ export function Modal(props: ModalProps) {
     animationType,
     presentationStyle,
     transparent,
+    detents,
     ...viewProps
   } = props;
   const { openModal, closeModal, addEventListener } = useModalContext();
   const [currentModalId, setCurrentModalId] = useState<string | undefined>();
   const navigation = useNavigation<NavigationProp<ParamListBase>>();
   useEffect(() => {
-    if (!areDetentsValid(props.detents)) {
-      throw new Error(`Invalid detents provided to Modal: ${JSON.stringify(props.detents)}`);
+    if (!areDetentsValid(detents)) {
+      throw new Error(`Invalid detents provided to Modal: ${JSON.stringify(detents)}`);
     }
-  }, [props.detents]);
+  }, [detents]);
```

Comment:
Detents are usually passed as inline array, so this effect and check will likely run on every render but guess its fine